gm4_enchantment_extractors/data/enchantment_extractors/functions/enchantment.mcfunction

```diff
@@ -0,0 +1,94 @@
+execute if score extract_chance gm4_random matches 3..7 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:silk_touch"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:silk_touch",lvl:1},{id:"minecraft:curse_of_vanishing",lvl:1}]}
+execute if score extract_chance gm4_random matches ..80 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:fortune"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:fortune",lvl:1}]}
+execute if score extract_chance gm4_random matches ..30 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:2s,id:"minecraft:fortune"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:fortune",lvl:2}]}
+execute if score extract_chance gm4_random matches ..5 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:3s,id:"minecraft:fortune"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:fortune",lvl:3}]}
+execute if score extract_chance gm4_random matches ..80 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:punch"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:punch",lvl:1}]}
+execute if score extract_chance gm4_random matches ..50 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:2s,id:"minecraft:punch"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:punch",lvl:2}]}
+execute if score extract_chance gm4_random matches ..75 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:lure"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:lure",lvl:1}]}
+execute if score extract_chance gm4_random matches ..65 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:2s,id:"minecraft:lure"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:lure",lvl:2}]}
+execute if score extract_chance gm4_random matches ..35 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:3s,id:"minecraft:lure"}]}}}] run replaceitem block ~ ~ ~ container.1 enchanted_book{StoredEnchantments:[{id:"minecraft:lure",lvl:3}]}
+execute if score extract_chance gm4_random matches ..8 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:curse_of_vanishing"}]}}}] run replaceitem block ~ ~ ~ container.2 enchanted_book{StoredEnchantments:[{id:"minecraft:curse_of_vanishing",lvl:1}]}
+execute if score extract_chance gm4_random matches ..5 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:mending"}]}}}] run replaceitem block ~ ~ ~ container.3 enchanted_book{StoredEnchantments:[{id:"minecraft:mending",lvl:1}]}
+execute if score extract_chance gm4_random matches ..5 if entity @s[nbt={Item:{tag:{Enchantments:[{lvl:1s,id:"minecraft:mending"}]}}}] run advancement grant @a[distance=..4] only gm4:enchantment_extractors_mending
```

Comment:
For consistency `grant @a[distance=..4]` should be `distance=..3` and a `gamemode=!spectator` should be added to avoid giving achievements to them.